siddhant/random-wallpaper-wallhaven.py

```diff
@@ -0,0 +1,31 @@
+import requests, bs4, os, random
+#import subprocess
+url = "https://alpha.wallhaven.cc/toplist"
+dir_name = 'wallpapers-wallhaven-toplist'
+if not os.path.exists(dir_name):
+    os.makedirs(dir_name)
+print("Downloading Page %s......" % url)
+res = requests.get(url)
+res.raise_for_status()
+soup = bs4.BeautifulSoup(res.text, "html.parser")
+top_images_list = soup.select('section ul li figure a')
+if top_images_list == []:
+    print('Could not find images!')
+else:
+    random_image_url = top_images_list[random.randint(1, len(top_images_list))].get('href')
+    res = requests.get(random_image_url)
+    res.raise_for_status()
+    soup = bs4.BeautifulSoup(res.text, 'html.parser')
```

Comment:
Why do you need to create another object? This has already been done in line number 10.

Reply:
The first time, the soup object downloads the whole top-images page, the next time, it downloads the view page for the randomly chosen wallpaper, and then traces to its original source, downloads it. I will refactor the variable names for clarity, my bad. :stuck_out_tongue:

Reply:
Actually, I feel that this logic can be improved further. :1st_place_medal:
Try to directly "get" the image URL by looking into the source code of the page and then save the image, and use it.

Reply:
I will get back on this ASAP

Reply:
Sure, you're doing fine. No hurries :)